Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreInnerStepFindBlocks.cs

```diff
@@ -39,43 +33,66 @@ public BlockStoreInnerStepFindBlocks(ILoggerFactory loggerFactory)
         public override async Task<InnerStepResult> ExecuteAsync(BlockStoreInnerStepContext context)
         {
             this.logger.LogTrace("()");
-            context.GetNextBlock();
-
-            if (await ShouldStopFindingBlocks(context))
-            {
-                if (!context.DownloadStack.Any())
-                {
-                    this.logger.LogTrace("(-):{0}", InnerStepResult.Stop);
-                    return InnerStepResult.Stop;
-                }
 
-                context.StopFindingBlocks();
-            }
-            else
+            while (await ShouldStopFindingBlocks(context) == false)
             {
-                context.BlockStoreLoop.BlockPuller.AskBlock(context.NextChainedBlock);
                 context.DownloadStack.Enqueue(context.NextChainedBlock);
-
-                if (context.DownloadStack.Count == context.BlockStoreLoop.BatchDownloadSize)
-                    context.StopFindingBlocks();
+                context.GetNextBlock();
             }
 
+            context.StopFindingBlocks();
+
+            if (context.DownloadStack.Any())
+                context.BlockStoreLoop.BlockPuller.AskForMultipleBlocks(context.DownloadStack.ToArray());
+
             this.logger.LogTrace("(-):{0}", InnerStepResult.Next);
+
             return InnerStepResult.Next;
         }
 
         private async Task<bool> ShouldStopFindingBlocks(BlockStoreInnerStepContext context)
         {
             this.logger.LogTrace("()");
 
-            bool res = (context.NextChainedBlock == null)
-                || (context.NextChainedBlock.Header.HashPrevBlock != context.InputChainedBlock.HashBlock)
-                || (context.NextChainedBlock.Height > context.BlockStoreLoop.ChainState.HighestValidatedPoW?.Height)
-                || (context.BlockStoreLoop.PendingStorage.ContainsKey(context.NextChainedBlock.HashBlock))
-                || (await context.BlockStoreLoop.BlockRepository.ExistAsync(context.NextChainedBlock.HashBlock));
+            if (context.NextChainedBlock == null)
+            {
+                this.logger.LogTrace("(-): {0} was null, return true", nameof(context.NextChainedBlock));
```

Comment:
@Aprogiena I honestly cannot see the benefit of adding logs to every IF statement here. If we want to know why a particular stop condition was reached for a particular case then we can use the debugger to step into the code and see what's going on. I'm going to take them out and just log the result.